### PR TITLE
qrtool: init at 0.8.4

### PR DESCRIPTION
**Pull request by philiptaron**
## Description of changes

qrtool is a command-line utility for encoding or decoding QR code.

Demonstration of qrtool

I added myself as maintainer, but I'm just a drive-by packager.

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/by-name/qr/qrtool/package.nix [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+  src = fetchFromGitHub {
+    owner = "sorairolake";
+    repo = "qrtool";
+    rev = "v${version}";
+    sha256 = "sha256-FoWUGhNfVILpYxmsnSzRIM1+R9/xFxCF7W1sdiHaAiA=";
+  };
+
+  cargoSha256 = "sha256-mtejnHCkN2krgFAneyyBpvbv5PZO3GigM2DJqrbHim4=";
+
+  meta = with lib; {
```
Comment (atalii) [inline pkgs/by-name/qr/qrtool/package.nix:20]: Could you add the changelog here? It looks like it's at `changelog = "https://github.com/sorairolake/qrtool/blob/v{version}/CHANGELOG.adoc";` but do double-check that. thanks!

**Review by atalii — CHANGES_REQUESTED** [review]
Looks mostly good! Do you think this is stable enough to want to backport to 23.05?

**Review by onemoresuza — CHANGES_REQUESTED** [review]
# Reviewed points

Result of `nixpkgs-review pr 266163` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>qrtool</li>
  </ul>
</details>

- [X] package path fits guidelines
- [X] package name fits guidelines
- [X] package version fits guidelines
- [X] package build on x86_64-linux
- [X] executables tested on x86_64-linux
- [X] `meta.description` is set and fits guidelines
- [X] `meta.license` fits upstream license
- [ ] `meta.platforms` is set\*1
- [X] `meta.maintainers` is set
- [X] build time only dependencies are declared in `nativeBuildInputs`
- [X] source is fetched using the appropriate function
- [X] the list of `phases` is not overridden

# Possible Improvements

1. When using `rustPlatform.buildRustPackage`, the `meta.platforms` attribute is already set --- see [here][build-rust-package-platforms]. Unless explicitly needed by the program --- *e. g.*, it does not build on some of those platforms --- there's no need to set it.

[build-rust-package-platforms]: <https://github.com/NixOS/nixpkgs/blob/3e99c3cd90572e0bb66078d59a0bda4fe05501a6/pkgs/build-support/rust/build-rust-package/default.nix#L155>

File: pkgs/by-name/qr/qrtool/package.nix [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+    platforms = platforms.unix;
```
Comment (onemoresuza) [inline pkgs/by-name/qr/qrtool/package.nix:21]:
```suggestion
```

File: pkgs/by-name/qr/qrtool/package.nix (hunk 1 above)
Comment (philiptaron) [reply pkgs/by-name/qr/qrtool/package.nix:20]: Done, thanks.

File: pkgs/by-name/qr/qrtool/package.nix (hunk 2 above)
Comment (philiptaron) [reply pkgs/by-name/qr/qrtool/package.nix]: Done, thanks!

**Comment by philiptaron** [comment]
> Looks mostly good! Do you think this is stable enough to want to backport to 23.05?

I don't know if "stable enough" is the bar for backports... it certainly doesn't meet my bar for importance to backport.

**Review by SuperSandro2000 — APPROVED** [review]
There shouldn't be any quotes around the package name in the PR title

**Review by atalii — APPROVED** [review]
Reasonable enough not to backport. LGTM!

**Review by onemoresuza — APPROVED** [review]

**Comment by philiptaron** [comment]
> There shouldn't be any quotes around the package name in the PR title

Thanks, updated.

**Pull request merged**